### PR TITLE
Adding support for 32 bit boxes (and 32 bit OSes)

### PR DESCRIPTION
**Pull request by rokaz**
Hi,
As discussed on Jabber, I send this PR for adding 32 bit boxes support and thus support for 32 bit OSes.

### CODE REVIEW

**Comment by rokaz** [comment]
And I forgot to add that these modifications imply to rename (or probably better to link) the 64 bit box files yunohost-$VERSION.box to yunohost-amd64-$VERSION.box on build.yunohost.org

**Comment by alexAubin** [comment]
Hi there,

did anyone test this ? I havent tested yet but it looks like something relatively easy to merge and useful for some people, so would be cool if we review it.

File: ynh-dev [hunk 1]
```diff
@@ -108,18 +111,30 @@ elif [ "$1" = "run" ]; then
...
-    elif [ "$VERSION" = "unstable" ]; then
-        BOX_NAME="yunohost/jessie-unstable"
-        BOX_URL="https://build.yunohost.org/yunohost-jessie-unstable.box"
+    elif [ "$VERSION" = "unstable64" ]; then
+        BOX_NAME="yunohost/jessie-amd64-unstable"
+        BOX_URL="https://build.yunohost.org/yunohost-jessie-amd64-unstable.box"
         IP="192.168.33.82"
+    elif [ "$VERSION" = "stable32" ]; then
+        BOX_NAME="yunohost/jessie-i386-stable"
+        BOX_URL="https://build.yunohost.org/yunohost-jessie-i386-stable.box"
```
Comment (M5oul) [inline ynh-dev]: Thoses six url do not exists.

Comment (alexAubin) [inline ynh-dev]: Yes.. @rokaz do you have the boxes somewhere ? We need somehow with access to build.yunohost.org

Comment (alexAubin) [inline ynh-dev]: (I just realized they are in https://github.com/YunoHost/Vagrantfile/pull/4 )

Comment (rokaz) [inline ynh-dev]: And I built them on my laptop also if needed, using the code in the PR you mentioned

File: ynh-dev [hunk 2]
```diff
@@ -99,7 +102,7 @@ elif [ "$1" = "create-env" ]; then
 elif [ "$1" = "run" ]; then
 
     DOMAIN=$2
-    VERSION='stable'
+    VERSION='stable64'
```
Comment (alexAubin) [inline ynh-dev]: I'm wondering if it's a good idea to rename these with "64" at the end. It kind of break retro-compatibility (or at least I think people are use to simply type "stable"/"testing"/"unstable"). I think simply having stable32/testing32/unstable32 for the particular 32-bit version would be good enough, no need to change the existing stable/testing/unstable.

Comment (opi) [inline ynh-dev]: I can't agree more, as 32bits will be slowly discontinued, and 64bits is de facto standard,

**Comment by alexAubin** [comment]
https://github.com/YunoHost/Vagrantfile/pull/4 has been merged.

I have no idea how the stuff on build.yunohost.org is supposed to appear (e.g. https://build.yunohost.org/yunohost-jessie-i386-stable.box) is supposed to be created. I guess we need a manual intervention on the server for this ?

**Comment by rokaz** [comment]
As far as I know, the images on build.yunohost.org were built manually (by opi?). I built myself them images for 32bits using the the script in https://github.com/YunoHost/Vagrantfile/prebuild, unfortunately I can't upload them because they are on a machine I don't have access anymore

File: ynh-dev (hunk 2 above)
Comment (rokaz) [inline ynh-dev]: I proposed this for the sake of consistency and predictability (in my head when the architecture is not specified, it is 32 bits but I probably live 10 years in the past :) )

**Comment by opi** [comment]
@rokaz you're right, I created & uploaded the .box files ; I guess I got to update them soon.

**Comment by Psycojoker** [comment]
Since https://github.com/YunoHost/ynh-dev/commit/5dc4bc6ce755424c5f3cb8a7fe0d8a86ecf84d48 I don't think that this PR is still valid :/

**Comment by alexAubin** [comment]
>Since 5dc4bc6 I don't think that this PR is still valid :/

It is ;) The PR of ljf was just meant to be able to run several separate VM (for the same version (e.g. unstable)). This PR adds 32 bits VMs.

**Comment by Psycojoker** [comment]
@zamentur all the explanation is here https://github.com/YunoHost/Vagrantfile/pull/5

Basically production was broken and ynh-dev unusable so I had to get back master to a working situation ASAP but had no time to implement the real fix. I've mentioned in this reverting PR but since we are overwhelmed by notification I'm not surprised you've missed it.

**Comment by alexAubin** [comment]
Closing this because we were already not really active on keeping the 64 bit boxes up to date and now we're moving to LXC...

**Pull request closed without merging**